third_party/txt/src/skia/paragraph_skia.cc

```diff
@@ -84,10 +89,105 @@ TextStyle SkiaToTxt(const skt::TextStyle& skia) {
   return txt;
 }
 
+class DisplayListParagraphPainter : public skt::ParagraphPainter {
+ public:
+  DisplayListParagraphPainter(DisplayListBuilder* builder,
+                              const std::vector<DlPaint>& dl_paints)
+      : builder_(builder), dl_paints_(dl_paints) {}
+
+  DlPaint toDlPaint(const DecorationStyle& decorStyle) {
+    DlPaint paint;
+    paint.setDrawStyle(DlDrawStyle::kStroke);
+    paint.setAntiAlias(true);
+    paint.setColor(decorStyle.getColor());
+    paint.setStrokeWidth(decorStyle.getStrokeWidth());
+    std::optional<DashPathEffect> dashPathEffect =
+        decorStyle.getDashPathEffect();
+    if (dashPathEffect) {
+      std::array<SkScalar, 2> intervals{dashPathEffect->fOnLength,
+                                        dashPathEffect->fOffLength};
+      paint.setPathEffect(
+          DlDashPathEffect::Make(intervals.data(), intervals.size(), 0));
+    }
+    return paint;
+  }
+
+  void drawTextBlob(const sk_sp<SkTextBlob>& blob,
+                    SkScalar x,
+                    SkScalar y,
+                    const SkPaintOrID& paint) override {
+    if (!blob) {
+      return;
+    }
+    size_t paint_id = std::get<PaintID>(paint);
+    FML_DCHECK(paint_id < dl_paints_.size());
+    builder_->drawTextBlob(blob, x, y, dl_paints_[paint_id]);
+  }
+
+  void drawTextShadow(const sk_sp<SkTextBlob>& blob,
+                      SkScalar x,
+                      SkScalar y,
+                      SkColor color,
+                      SkScalar blurSigma) override {
+    if (!blob) {
+      return;
+    }
+    DlPaint paint;
+    paint.setColor(color);
+    DlBlurMaskFilter filter(SkBlurStyle::kNormal_SkBlurStyle, blurSigma, false);
```

Comment:
Check blurSigma for non-zero first. (Skia uses `!= 0.0` but then their `MakeBlur` factory returns null if sigma isn't > 0, so I'd just use `> 0.0` here).

Reply:
In this case we will always have a MaskFilter, but it may be a NOP. Didn't this cause any performance regressions?

Reply:
Added this check.

AFAIK we do not have a benchmark focused on text shadows.  I did not notice any changes in general benchmarks due to this PR.  But text shadows are rarely used and would not likely show up there.